pkg/awsclient/awsclient.go

```diff
@@ -0,0 +1,85 @@
+package awsclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	awsclient "github.com/aws/aws-sdk-go/aws/client"
+	"github.com/go-logr/logr"
+	capa "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/scope"
+	capiutil "sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/capa-iam-controller/pkg/key"
+)
+
+type AWSClientConfig struct {
+	ClusterName string
+	CtrlClient  client.Client
+	Log         logr.Logger
+}
+
+type AwsClient struct {
+	clusterName string
+	ctrlClient  client.Client
+	log         logr.Logger
+}
+
+func New(config AWSClientConfig) (*AwsClient, error) {
+	if config.ClusterName == "nil" {
```

Comment:
Is this supposed to be  `"nil"` or just `""`?

Reply:
ah, nice catch, should be `""`